domains/certificates/Query.API/API/DataSyncSyncer/Service/DataSync.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using API.DataSyncSyncer.Service.Configurations;
+using API.MasterDataService;
+using CertificateEvents;
+using CertificateEvents.Primitives;
+using EnergyOriginDateTimeExtension;
+using IdentityModel.Client;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+
+namespace API.DataSyncSyncer.Service.Datasync;
+
+public class DataSync : IDataSync
+{
+    private readonly HttpClient httpClient;
+    private readonly DatasyncOptions options;
+    private readonly ILogger<DataSync> logger;
+
+    public DataSync(HttpClient httpClient, IOptions<DatasyncOptions> options, ILogger<DataSync> logger)
+    {
+        this.logger = logger;
+        this.httpClient = httpClient;
+        this.options = options.Value;
+    }
+
+    public async Task<EnergyMeasuredIntegrationEvent> GetMeasurement(string gsrn, Period period)
+    {
+        logger.LogInformation(
+            "Fetching data in period from {from} to: {to}", period.DateFrom, period.DateTo
+        );
+
+        var url = $"{options.Url}/measurements?gsrn={gsrn}&dateFrom={period.DateFrom}&dateTo={period.DateTo}";
```

Comment:
It is also possible to set the BaseAddress on the HttpClient. Then you do not have to prepend it here. Furthermore, this will also fix if someone in the config ends the url with a "/"